components/support/migration/src/main/java/mozilla/components/support/migration/FennecMigrator.kt

```diff
@@ -597,22 +603,27 @@ class FennecMigrator private constructor(
 
             versionedMigration.migration.metricTotalDuration().start()
 
-            val migrationResult = when (versionedMigration.migration) {
-                Migration.History -> migrateHistory()
-                Migration.Bookmarks -> migrateBookmarks()
-                Migration.OpenTabs -> migrateOpenTabs()
-                Migration.FxA -> migrateFxA()
-                Migration.Gecko -> migrateGecko(versionedMigration.version)
-                Migration.Logins -> migrateLogins()
-                Migration.Settings -> migrateSharedPrefs()
-                Migration.Addons -> migrateAddons()
-                Migration.TelemetryIdentifiers -> migrateTelemetryIdentifiers()
-                Migration.SearchEngine -> migrateSearchEngine()
-                Migration.PinnedSites -> migratePinnedSites()
+            val migrationResult: Result<*> = try {
+                when (versionedMigration.migration) {
+                    Migration.History -> migrateHistory()
+                    Migration.Bookmarks -> migrateBookmarks()
+                    Migration.OpenTabs -> migrateOpenTabs()
+                    Migration.FxA -> migrateFxA()
+                    Migration.Gecko -> migrateGecko(versionedMigration.version)
+                    Migration.Logins -> migrateLogins()
+                    Migration.Settings -> migrateSharedPrefs()
+                    Migration.Addons -> migrateAddons()
+                    Migration.TelemetryIdentifiers -> migrateTelemetryIdentifiers()
+                    Migration.SearchEngine -> migrateSearchEngine()
+                    Migration.PinnedSites -> migratePinnedSites()
+                }
+            } catch (e: Exception) {
+                logger.error("Unexpected error while migrating $versionedMigration", e)
+                Result.Failure<Any>(e)
+            } finally {
+                versionedMigration.migration.metricTotalDuration().stop()
             }
 
-            versionedMigration.migration.metricTotalDuration().stop()
```

Comment:
If we measured total duration, did we see this drop off consistently with the funnel?